pkg/nodeconfig/nodeconfig.go

```diff
@@ -172,14 +173,12 @@ func (nc *nodeConfig) Configure() error {
 
 		// Ensure we are annotating the node as soon as the Node object is created, so that we can identify which
 		// controller should be watching it
-		nc.addAdditionalAnnotations()
-		nc.addPubKeyHashAnnotation()
-		node, err := nc.k8sclientset.CoreV1().Nodes().Update(context.TODO(), nc.node, meta.UpdateOptions{})
-		if err != nil {
-			return errors.Wrapf(err, "error updating public key hash and additional annotations on node %s",
-				nc.node.GetName())
+		if err := nc.addAdditionalAnnotations(); err != nil {
+			return errors.Wrapf(err, "error updating additional annotations on node %s", nc.node.GetName())
+		}
+		if err := nc.addAnnotation(PubKeyHashAnnotation, nc.publicKeyHash); err != nil {
+			return errors.Wrapf(err, "error updating public key hash on node %s", nc.node.GetName())
```

Comment:
Can this be done as one operation? I'd think we'd like to avoid increasing the amount of API calls as much as possible